web/modules/custom/dpl_login/dpl_login.module

```diff
@@ -71,41 +78,48 @@ function dpl_login_openid_connect_pre_authorize(AccountInterface|FALSE $account,
   $api_factory = \Drupal::service('dpl_fbs.api_factory');
   $api = $api_factory->getPatronApi($token->token);
 
-  try {
-    $patron = $api->getPatronInformationByPatronIdV2();
-    $patron_block_status = $patron->getPatron()?->getBlockStatus() ?? [];
-    $patron_blocked_reasons = array_map(function (BlockStatus $block_status) use ($logger) {
-      try {
-        return BlockedReason::from($block_status->getBlockedReason());
-      }
-      catch (UnexpectedValueException $e) {
-        $logger->warning('Unable to map blocked reason @reason: @message', [
-          '@reason' => $block_status->getBlockedReason(),
-          '@message' => $e->getMessage(),
-        ]);
-        return BlockedReason::UNKNOWN();
-      }
-    }, $patron_block_status);
-
-    // Only a subset of the blocked reasons should block login.
-    $deny_login_reasons = [
-      BlockedReason::DECEASED(),
-      BlockedReason::ACCOUNT_STOLEN(),
-      BlockedReason::BLOCKED_FROM_SELFSERVICE(),
-    ];
-    $patron_deny_reasons = array_intersect($patron_blocked_reasons, $deny_login_reasons);
-
-    // Return FALSE to block login if the patron block reasons match one or
-    // more of the deny reasons. The OpenID Connect module will provide
-    // feedback.
-    return !(count($patron_deny_reasons) > 0);
-  }
-  catch (ApiException $e) {
-    $logger->error('Authorization blocked. Unable to get patron information: @message', [
-      '@message' => $e->getMessage(),
-    ]);
-    return FALSE;
+  // Check if the user is associated with any agencies.
+  // If that is the case, the user is new, and we skip
+  // checking for blocked status.
+  if (!empty($context['userinfo']['attributes']['agencies'])) {
+    try {
```

Comment:
Please flip the logic so you do not have to indent the whole block of code.
So this should do it, I think:
```php
if (empty($context['userinfo']['attributes']['agencies'])) {
  return TRUE;
}
```